src/lib/falcor/falcorUtils.js

```diff
@@ -40,95 +40,110 @@ export function validateFalcorPathSets(falcorPathSets) {
   return falcorPathSets;
 }
 
-function recurseOnCache(curObject, remainingKeySets) {
-  if (remainingKeySets.length === 0) {
-    return true;
-  }
-  const nextRemainingKeySets = remainingKeySets.slice(1);
-  let nextKeySet = remainingKeySets[0];
-  if (!(nextKeySet instanceof Array)) {
-    nextKeySet = [nextKeySet];
+function followPath(path, object) {
+  // If using dot notation obj.key.key.key
+  if (typeof path === "string") {
+    path = path.split('.');
   }
-  return nextKeySet.every((key) => {
-    if (key !== null && typeof key === "object") {
-      let start = 0;
-      if (key.hasOwnProperty("from")) {
-        start = key.from;
-      }
-      let end;
-      if (key.hasOwnProperty("to")) {
-        if (key.hasOwnProperty("length")) {
-          throw new Error("Falcor Range cannot have both 'to' and 'length' properties at falcor KeySet: " + JSON.stringify(key));
+  return path.reduce((currentObject, nextChild) => {
+    if (currentObject !== undefined && currentObject.hasOwnProperty(nextChild)) {
+      return currentObject[nextChild];
+    }
+
+    return undefined;
+  }, object);
+}
+
+export function pathSetsInCache(cache, falcorPathSets) {
+  function handleCheckingSingleKey(curObject, nextRemainingKeySets, key) {
```

Comment:
The only part of this that is confusing is what is the `nextRemainingKeySets` argument here.